src/watchmaker/workers/salt.py

```diff
@@ -198,6 +204,14 @@ def before_install(self):
                 "The 'ou_path' option is required, but not provided."
             )
 
+        if self.computer_name and self.computer_name_pattern:
+            if not re.match(self.computer_name_pattern + r"\Z",
+                            self.computer_name):
```

Comment:
```suggestion
            if not re.match(self.computer_name_pattern, self.computer_name):
```

Reply:
Do we know it still works correctly if the user specifies a full match pattern themselves? e.g. `(?i)^xyz[\d]{3}[a-z]{8}[ex]$`

Reply:
Yes, it still works, I have tested it but will add some tests proving it.